core/io/file_access.cpp

```diff
@@ -258,7 +258,11 @@ String FileAccess::fix_path(const String &p_path) const {
 		case ACCESS_RESOURCES: {
 			if (ProjectSettings::get_singleton()) {
 				if (r_path.begins_with("uid://")) {
-					r_path = ResourceUID::uid_to_path(r_path);
+					if (ResourceUID::has_uid(r_path)) {
+						r_path = ResourceUID::uid_to_path(r_path);
```

Comment:
```suggestion
					const ResourceUID::ID id = ResourceUID::get_singleton()->text_to_id(r_path);
					if (ResourceUID::has_id(id)) {
						r_path = ResourceUID::get_id_path(id);
```
The new method results in multiple conversions, which can be easily avoided.